diff --git a/projects/common/01bsp_radio/01bsp_radio.c b/projects/common/01bsp_radio/01bsp_radio.c
index f7576ebb..a2482f69 100644
--- a/projects/common/01bsp_radio/01bsp_radio.c
+++ b/projects/common/01bsp_radio/01bsp_radio.c
@@ -27,8 +27,10 @@ end of frame event), it will turn on its error LED.
 
 #define LENGTH_PACKET   125+LENGTH_CRC  ///< maximum length is 127 bytes
 #define LEN_PKT_TO_SEND 20+LENGTH_CRC
-#define CHANNEL         11            ///< 11=2.405GHz
-#define TIMER_PERIOD    (0xffff>>4)    ///< 0xffff = 2s@32kHz
+//#define CHANNEL         11            ///< 11=2.405GHz
+//#define TIMER_PERIOD    (0xffff>>8)    ///< 0xffff = 2s@32kHz
+#define CHANNEL         13            ///< 11=2.405GHz
+#define TIMER_PERIOD    (0xffff)    ///< 0xffff = 2s@32kHz
 #define ID              0x99           ///< byte sent in the packets
 
 uint8_t stringToSend[]  = "+002 Ptest.24.00.12.-010\n";
@@ -270,13 +272,19 @@ int mote_main(void) {
                     // prepare packet
                     app_vars.packet_len = sizeof(app_vars.packet);
                     i = 0;
-                    app_vars.packet[i++] = 't';
-                    app_vars.packet[i++] = 'e';
-                    app_vars.packet[i++] = 's';
-                    app_vars.packet[i++] = 't';
-                    app_vars.packet[i++] = CHANNEL;
-                    while (i<app_vars.packet_len) {
-                        app_vars.packet[i++] = ID;
+                    //app_vars.packet[i++] = 't';
+                    //app_vars.packet[i++] = 'e';
+                    //app_vars.packet[i++] = 's';
+                    //app_vars.packet[i++] = 't';
+                    //app_vars.packet[i++] = CHANNEL;
+                    //while (i<app_vars.packet_len) {
+                    //    app_vars.packet[i++] = ID;
+                    //}
+                    
+                    const char* message = "IOTA5109_TEST";
+                    while (i < strlen(message)) {
+                        app_vars.packet[i] = message[i];
+                        i++;
                     }
 
                     // start transmitting packet
diff --git a/projects/nrf52840_dk/nrf52840_dk.emProject b/projects/nrf52840_dk/nrf52840_dk.emProject
index 83da06de..9a50509c 100644
--- a/projects/nrf52840_dk/nrf52840_dk.emProject
+++ b/projects/nrf52840_dk/nrf52840_dk.emProject
@@ -705,18 +705,14 @@
       </file>
     </folder>
     <folder Name="chips">
-      <file file_name="01bsp_bmx160/bmp388.c">
-        <configuration Name="Debug" build_exclude_from_build="No" />
-      </file>
-      <file file_name="01bsp_bmx160/bmp388.h">
-        <configuration Name="Debug" build_exclude_from_build="No" />
-      </file>
       <file file_name="../../bsp/chips/bmp388/b388.c">
         <configuration Name="Debug" build_exclude_from_build="Yes" />
       </file>
       <file file_name="b388.h">
         <configuration Name="Debug" build_exclude_from_build="Yes" />
       </file>
+      <file file_name="../../bsp/chips/bmp388/bmp388.h" />
+      <file file_name="../../bsp/chips/bmp388/bmp388.c" />
     </folder>
     <folder Name="inc">
       <file file_name="af.h" />
diff --git a/projects/nrf52840_dk/nrf52840_dk.emSession b/projects/nrf52840_dk/nrf52840_dk.emSession
index afaa381c..e3fbcabf 100644
--- a/projects/nrf52840_dk/nrf52840_dk.emSession
+++ b/projects/nrf52840_dk/nrf52840_dk.emSession
@@ -97,7 +97,7 @@
   <SessionOpenFile windowGroup="DockEditLeft" x="0" y="29" useTextEdit="1" openedFrom="W:/HKUST_PhD_Y1/IOTA5901/openwsn-fw/projects/nrf52840_dk/01bsp_bmp388/01bsp_bmp388.c" path="../../bsp/boards/nrf52840_dk/board.c" left="0" top="0" codecName="Default"/>
   <SessionOpenFile windowGroup="DockEditLeft" x="0" y="116" useTextEdit="1" openedFrom="W:/HKUST_PhD_Y1/IOTA5901/openwsn-fw/projects/nrf52840_dk/01bsp_bmp388/01bsp_bmp388.c" path="../../bsp/boards/nrf52840_dk/uart.c" left="0" top="99" codecName="Default"/>
   <SessionOpenFile windowGroup="DockEditLeft" x="19" y="11" useTextEdit="1" path="../common/01bsp_i2c/01bsp_i2c.c" left="0" top="7" codecName="Default"/>
-  <SessionOpenFile windowGroup="DockEditLeft" x="41" y="70" useTextEdit="1" path="../common/01bsp_radio/01bsp_radio.c" left="0" selected="1" top="33" codecName="Default"/>
+  <SessionOpenFile windowGroup="DockEditLeft" x="41" y="70" useTextEdit="1" path="../common/01bsp_radio/01bsp_radio.c" left="0" selected="1" top="120" codecName="Default"/>
  </Files>
- <EMStudioWindow activeProject="01bsp_radio" fileDialogDefaultFilter="*" autoConnectTarget="J-Link" buildConfiguration="Debug" sessionSettings="" debugSearchFileMap="" fileDialogInitialDirectory="W:/HKUST_PhD_Y1/IOTA5901/openwsn-fw/projects/nrf52840_dk/01bsp_bmx160" debugSearchPath="" autoConnectCapabilities="3711"/>
+ <EMStudioWindow activeProject="01bsp_radio" fileDialogDefaultFilter="*" autoConnectTarget="J-Link" buildConfiguration="Debug" sessionSettings="" debugSearchFileMap="" fileDialogInitialDirectory="W:/HKUST_PhD_Y1/IOTA5901/openwsn-fw/projects/nrf52840_dk/01bsp_bmx160" debugSearchPath="" autoConnectCapabilities="3199"/>
 </session>
